ci-operator/step-registry/cucushift/installer/check/gcp/private/cucushift-installer-check-gcp-private-commands.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+GCP_REGION="${LEASED_RESOURCE}"
+
+GCP_BASE_DOMAIN="$(< ${CLUSTER_PROFILE_DIR}/public_hosted_zone)"
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/gce.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+export KUBECONFIG=${SHARED_DIR}/kubeconfig
+
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+    # shellcheck disable=SC1090
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+ret=0
+
+echo "INFO: (1/4) Checking publish policy..."
+publish_policy=$(oc get configmap -n kube-system cluster-config-v1 -oyaml | grep 'publish:')
+echo "INFO: ${publish_policy}"
+if [[ "${publish_policy}" =~ "publish: Internal" ]]; then
+    echo "INFO: Publish policy check passed."
+else
+    echo "ERROR: Publish policy check failed."
+    ret=$(( $ret | 1 ))
+fi
+
+echo "INFO: (2/4) Checking if any dns record-sets in the base domain (public zone)..."
+base_domain_zone_name=$(gcloud dns managed-zones list --filter="dnsName=${GCP_BASE_DOMAIN}." --format="table(name)" | grep -v NAME)
+echo "INFO: base domain zone name '${base_domain_zone_name}'"
+gcloud dns record-sets list --zone "${base_domain_zone_name}" | grep "${CLUSTER_NAME}" && ret=$(( $ret | 2 ))
```

Comment:
if the grep command did not find out any matched result, the whole script would exit directly, right?

Reply:
Here I use "OR" so that all checks would be executed, and the exit code could tell which check(s) failed.

Reply:
In this script, `set -o errexit` is enabled, once any command exit with non-zero, the whole script would exit as failure, but not execute the following check any more.
```
[root@preserve-jialiu ~]# gcloud dns record-sets list --zone "qe" |grep "jialiu-non"; echo $?
1
```
So that means, once the above `grep` command did not find out any matched result, the 43 line will make the whole script exit as failure.

Reply:
Or do you want :
```
gcloud dns record-sets list --zone "${base_domain_zone_name}" | grep "${CLUSTER_NAME}" || ret=$(( $ret | 2 ))
```
but not:
```
gcloud dns record-sets list --zone "${base_domain_zone_name}" | grep "${CLUSTER_NAME}" && ret=$(( $ret | 2 ))
```

Reply:
Okay, ignore the above comment, sounds `&& ret=$(( $ret | 2 ))` help avoid the direct exit.